### ISSUE TITLE
Replace lua_pushnumber() with lua_pushinteger()?

### ISSUE THREAD

**Issue by Josef-Friedrich** [ufyTeX/luaharfbuzz#58]
```lua
#! luahbtex --luaonly

local face = luaharfbuzz.Face.new('/usr/share/fonts/truetype/noto/NotoSans-Regular.ttf')
if face == nil then error("Font not found!") end

for _, char in ipairs(face:collect_unicodes()) do
  print(char, unicode.utf8.char(char))
end
```

When I run the above script I get this output:

```
0.0	
13.0	
32.0	 
33.0	!
34.0	"
```

I would have expected this output

```
0	
13	
32	 
33	!
34	"
```

```c
 lua_pushinteger(L, ++i); 
 lua_pushinteger(L, c); 
```

instead of ?

https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L275-L276

**Comment by deepakjois** [comment]
Sorry, I haven't looked at this in a long time. Happy to merge a pull request that will pass all tests.

**Comment by deepakjois** [comment]
Closed by #59 ?
